### PR TITLE
Ugudip resource version call chain branch

### PR DESCRIPTION
**Pull request by ugudip**
Changes:

1. Removed the callchain initiator and rebind model changes and replaced it with simple progress chain logic to prevent call chain being overused in the predict Arn method.
2. Added Unit tests to the made changes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java [hunk 1]
```diff
@@ -47,88 +45,77 @@
...
-            if (!response.isSuccess()) {
-                return ProgressEvent.failed(
-                        model, context, response.getErrorCode(), response.getMessage());
+            //make a ListTypeVersions API call to get all versionIDs for status LIVE for the first time
+            final ListTypeVersionsResponse listTypeVersionsResponse;
+
+            try {
+                listTypeVersionsResponse = proxyClient.injectCredentialsAndInvokeV2(
+                        Translator.translateToListTypeVersionsRequest(model, marker, context.getDeprecatedStatus()), proxyClient.client()::listTypeVersions);
+            } catch (CfnRegistryException e) {
```
Comment (ammokhov) [inline aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java:63]: what does this exception mean?

File: aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java [hunk 2]
```diff
@@ -47,88 +45,77 @@
...
-                        } else {
-                            return latestVersion < first ? summary : currentLatest;
-                        }
-                    }
-            ).orElse(latest);
-            if (callResponse.nextToken() != null) {
-                listApi = listApi.rebindModel(callResponse);
+            latest = getLatestVersion(latest, currentLatest, listTypeVersionsResponse);
+            marker = listTypeVersionsResponse.nextToken();
+            if (StringUtils.isNullOrEmpty(marker) && context.getDeprecatedStatus() == DeprecatedStatus.LIVE) {
```
Comment (ammokhov) [inline aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java:77]: if token is null it means the pages have been exhausted

File: aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java [hunk 3]
```diff
@@ -47,88 +45,77 @@
...
             }
-
-            String arn = latest.arn();
-            String currentArn = currentLatest.arn();
-            Integer currentLatestVersion = Integer.valueOf(currentArn.substring(currentArn.lastIndexOf("/") + 1));
-            Integer latestVersion = Integer.valueOf(arn.substring(arn.lastIndexOf("/") + 1));
-            Integer current = currentLatestVersion > latestVersion ? currentLatestVersion : latestVersion; // to get the latest(largest) version ID
-            arn = arn.substring(0, arn.lastIndexOf("/") + 1)
-                    .concat(String.format("%08d", current + 1));
+            String arn = getCurrentArn(latest, currentLatest); //returns the largest version ID +1
```
Comment (ammokhov) [inline aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java:82]: not sure Im following why u need this

File: aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java (hunk 1 above)
Comment (ugudip) [reply aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java:63]: The ListTypeVersions API throws CfnRegistryException when the type is not yet registered. In this case the predicted version id will be 0000001.

File: aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java (hunk 2 above)
Comment (ugudip) [reply aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java:77]: yes, but we need to check for each status - LIVE and DEPRECATED. The default is LIVE and if the token is exhausted for LIVE, we need to switch it to DEPRECATED and check for the latest verison ID.

File: aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java (hunk 3 above)
Comment (ugudip) [reply aws-cloudformation-resourceversion/src/main/java/software/amazon/cloudformation/resourceversion/CreateHandler.java:82]: This method decides the final predicted Arn of the version. The latest and the currentLatest both are the maximum versions for Live and deprecated status respectively. We need to extract the Arn from these TypeVersionSummaries by picking the largest version and incrementing 1.

**Pull request closed without merging**